### ISSUE TITLE
High latency while doing concurrent unary rpc over multi connection

### ISSUE THREAD

**Issue by ohs30359-nobuhara** [grpc/grpc-node#863]
### Problem description
Latency of unary RPC when connecting multiple client connections gets worse

### Environment
os: mac os 
node version: v10.11.0

test tool: [GitHub - bojand/ghz: Simple gRPC benchmarking and load testing tool](https://github.com/bojand/ghz)
code: [GitHub - ohs30359-nobuhara/grpc-sample](https://github.com/ohs30359-nobuhara/grpc-sample)

### Additional context
The application we are testing is a simple server that returns a fixed rpc.
Latency will be degraded for connect multiple clients to the server.

It is the result of loadtest now.

```
[ client connection 1, total request 100 ]
10% in 0.32 ms
25% in 0.35 ms
50% in 0.40 ms
75% in 0.48 ms
90% in 0.61 ms
95% in 0.74 ms
99% in 1.55 ms

[ client connection 5, total request 500 ]
10% in 1.01 ms
25% in 1.18 ms
50% in 1.41 ms
75% in 1.81 ms
90% in 2.51 ms
95% in 3.17 ms
99% in 6.46 ms

[ client connection 10, total request 1000 ]
10% in 2.63 ms
25% in 2.87 ms
50% in 3.21 ms
75% in 3.81 ms
90% in 4.73 ms
95% in 5.47 ms
99% in 35.72 ms
```

command
```
ghz --insecure \
  --proto ./protos/service.proto \
  --call test_user.UserDomain.getUsers \
  -d '{"resultCount": 1}' \
  --connections 10 \
  -c 10 \
  -n 500 \
  0.0.0.0:50051
```

server code
https://github.com/ohs30359-nobuhara/grpc-sample/blob/master/src/server.ts

I understand that if the connection increases, it will deteriorate
Will the performance deteriorate so far?

**Comment by nicolasnoble** [comment]
Please share your server code.

**Comment by nicolasnoble** [comment]
Also, please fill the required fields from the template, indicating your version of grpc, platform, etc.

**Comment by ohs30359-nobuhara** [comment]
I forgot to list the code
repository is here
[GitHub-ohs30359-nobuhara / grpc-sample] (https://github.com/ohs30359-nobuhara/grpc-sample)

> Also, please fill the required fields from the template, indicating your version of grpc, platform, etc.

Sorry, I do not know how to check the grpc version ...
Since the grpc code is generated using "grpc_tools_node_protoc_ts", is this version applicable?

**Comment by nicolasnoble** [comment]
Huh, you're not using grpc directly, but a third party package... interesting.

https://github.com/agreatfool/grpc_tools_node_protoc_ts

**Comment by ohs30359-nobuhara** [comment]
Am I wrong to ask a question?
This question is https://github.com/agreatfool/grpc_tools_node_protoc_ts
Should I?

**Comment by murgatroid99** [comment]
It's a little hard to follow what your code is doing. You have grpc-gateway code and Kubernetes and Envoy configurations. Are all of those part of the loadtest you're doing? Can you reproduce the same problem with just Node code?

In general, if you have a single server you should expect some slowdown with many concurrent requests. The node server is single-threaded and handles incoming calls asynchronously, so as the number of concurrent requests increases, the server does more work on other calls between receiving and handling any particular call.

**Comment by ohs30359-nobuhara** [comment]
I'm sorry
This was bad for my test

Since it was executed on a local machine, CPU and Memory were exhausted and performance was degraded.

Please close this issue ...
